### PR TITLE
all the basics od the language focusing on difference rather than sim…

### PR DESCRIPTION
**Pull request by corinnekrych**
…ilitudes
@edewit is it ok? do I miss sth?

### CODE REVIEW

File: Basics.workbook [hunk 1]
```diff
@@ -88,9 +50,127 @@ class Rectangle
...
+string emptyString = String.Empty;
+string anotherEmptyString = "";
+```
+* String formating / String interpolation
+
+```csharp
+int x = 1, y = 2;
+int sum = x + y;
+string sumCalculation = String.Format("{0} + {1} = {2}", x, y, sum);
+Console.WriteLine(sumCalculation);
```
Comment (edewit) [inline Basics.workbook:127]: let's also add the string interpolation form:

``` csharp
$"{x} + {y} = {sum}"
```

File: Basics.workbook [hunk 2]
```diff
@@ -88,9 +50,127 @@ class Rectangle
...
+Console.WriteLine("Found orange in position: " + fruit.IndexOf("orange"));
+Console.WriteLine("Found lemon in position: " + fruit.IndexOf("lemon"));
+```
+
+# Arrays, List and Dictionnaries 
+
+Similar to Java:
+
+```csharp
+List<string> fruits = new List<string>();
```
Comment (edewit) [inline Basics.workbook]: Let's use `IList<string>` here like one does in Java

File: Basics.workbook [hunk 3]
```diff
@@ -88,9 +50,127 @@ class Rectangle
...
+// add fruits
+fruits.Add("apple");
+fruits.Add("banana");
+fruits.Add("orange");
+
```
Comment (edewit) [inline Basics.workbook:156]: and here let's show that one can do this also with a collection initializer:

``` csharp
var fruits = new List<string> {"apple", "banana", "orange"};
```

File: Basics.workbook [hunk 4]
```diff
@@ -88,9 +50,127 @@ class Rectangle
...
+// now remove the banana
+fruits.RemoveAt(1);
+Console.WriteLine(fruits.Count);
+
+```
+
+```csharp
+Dictionary<string, long> phonebook = new Dictionary<string, long>();
```
Comment (edewit) [inline Basics.workbook]: same here lets use `IDictionary` instead

File: Basics.workbook [hunk 5]
```diff
@@ -88,9 +50,127 @@ class Rectangle
...
+phonebook.Add("Alex", 415434543);
+phonebook["Jessica"] = 415984588;
+foreach (KeyValuePair<string, long> item in phonebook)
+{
+    Console.WriteLine(item.Key + "   " + item.Value);
+}
+phonebook.Remove("Jessica");
+Console.WriteLine(phonebook.Count);
+
```
Comment (edewit) [inline Basics.workbook:173]: Add the example using collection initializer:

``` csharp
var phonebook = new Dictionary<string, long> {{"Alex", 415434543}, {"Jessica", 415984588}};
```

**Comment by corinnekrych** [comment]
@edewit ok to merge?

**Pull request merged**